google-cloud-datastore/src/test/java/com/google/cloud/datastore/DatastoreTest.java

```diff
@@ -191,6 +192,8 @@ public void setUp() {
 
   @AfterClass
   public static void afterClass() throws IOException, InterruptedException, TimeoutException {
+    // TODO(gapic_upgrade): Temporarily addressing the flaky connection refused error
+    EmulatorUtils.checkHealth(helper.getPort());
```

Comment:
apologies, I know we had some discussion around this but on second (third? :)) thought, why not just add a MultipleAttemptsRule to DatastoreTest so the test will retry on flakes? example: https://github.com/googleapis/java-datastore/blob/main/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java#L143

Reply:
nice idea. let me try it out quickly.